test/widgets/chat/chat_page_test.dart

```diff
@@ -0,0 +1,176 @@
+      final textField = find.descendant(of: messageBar, matching: find.byType(TextFormField));
+      expect(textField, findsOneWidget);
+      await tester.enterText(textField, 'Hello World');
+      await tester.tap(find.byType(IconButton));
+      verifyRequest(
+        processor,
+        urlMatcher: equals('/rest/v1/messages'),
+        methodMatcher: equals('POST'),
+        bodyMatcher: equals({
+          'chat_id': chatId,
+          'content': 'Hello World',
+          'sender_id': 2,
+        }),
+      ).called(1);
+    });
+  });
+  group('ChatBubbles', () {
+    testWidgets('are shown', (WidgetTester tester) async {
+      whenRequest(processor).thenReturnJson([
+        MessageFactory().generateFake().toJsonForApi(),
+        MessageFactory().generateFake().toJsonForApi(),
+        MessageFactory().generateFake().toJsonForApi(),
+        MessageFactory().generateFake().toJsonForApi(),
+        MessageFactory().generateFake().toJsonForApi(),
+        MessageFactory().generateFake().toJsonForApi(),
+      ]);
+      await pumpMaterial(tester, ChatPage(profile: profile, chatId: chatId));
+      await tester.pump();
+      expect(find.byType(ChatBubble), findsNWidgets(6));
+    });
+    testWidgets('are not when there are no messages', (WidgetTester tester) async {
+      whenRequest(processor).thenReturnJson([]);
+      await pumpMaterial(tester, ChatPage(profile: profile, chatId: chatId));
+      await tester.pump();
+      expect(find.byType(ChatBubble), findsNothing);
+    });
+    testWidgets('shows Icon.done_all when message is from current user', (WidgetTester tester) async {
+      whenRequest(processor).thenReturnJson([
+        MessageFactory().generateFake(senderId: profile.id).toJsonForApi(),
+      ]);
+      SupabaseManager.setCurrentProfile(profile);
+      await pumpMaterial(tester, ChatPage(profile: profile, chatId: chatId));
+      await tester.pump();
+      final Finder chatBubble = find.byType(ChatBubble);
+      expect(chatBubble, findsOneWidget);
+      expect(find.descendant(of: chatBubble, matching: find.byIcon(Icons.done_all)), findsOneWidget);
+    });
+    testWidgets('shows Icon.done_all not when message is from other user', (WidgetTester tester) async {
+      whenRequest(processor).thenReturnJson([
+        MessageFactory().generateFake(senderId: profile.id! + 1).toJsonForApi(),
+      ]);
+      await pumpMaterial(tester, ChatPage(profile: profile, chatId: chatId));
+      await tester.pump();
+      final Finder chatBubble = find.byType(ChatBubble);
+      expect(chatBubble, findsOneWidget);
+      expect(find.descendant(of: chatBubble, matching: find.byIcon(Icons.done_all)), findsNothing);
+    });
+    testWidgets('shows tail iff it should be shown', (WidgetTester tester) async {
+      final Profile currentProfile = ProfileFactory().generateFake(id: 2);
+      SupabaseManager.setCurrentProfile(currentProfile);
+      final List<Map<String, dynamic>> messages = [
```

Comment:
Not sure I like this test. Maybe the messages should each get a descriptive name that explains why they are not shown?

Reply:
i added something not exactly sure what to ride

Reply:
*write xD

Reply:
Yeah I would have done
```
messageForCurrentUser = ...;
otherMessageForCurrentUser = ...;
newestMessageForCurrentUser = ...;
...
messages = [messageForCurrentUser, otherMessageForCurrentUser, newestMessageForCurrentUser, ...]

But I guess your way is also fine.